### ISSUE TITLE
TypeError: Cannot read property 'find' of null (layout.vue:95)

### ISSUE THREAD

**Issue by allanleonardjr** [directus/app#2940]
**Version:** 8.8.1
**Server, OS & Database Details:** Azure Linux Web App, MySQL
**Error Message:** `TypeError: Cannot read property 'find' of null`

![image](https://user-images.githubusercontent.com/16966049/89697774-ae375f00-d8eb-11ea-98b0-6fb0878bef0a.png)
![image](https://user-images.githubusercontent.com/16966049/89697821-ee96dd00-d8eb-11ea-8bdb-b88433100f41.png)
![image](https://user-images.githubusercontent.com/16966049/89698297-69f98e00-d8ee-11ea-90cc-9a1f60db65d7.png)

**Steps to Reproduce:**
* This happens whenever I navigate to the File Library (/public/admin/#/{project}/files)

**Relevant Info:**
* There are 597 files in the Library (I manually migrated them over from a v6 install)
* File Types Used: .bmp, .css, .docx, .eps, .gif, .jpg, .msi, .pdf, .png, .pptx, .ps1, .rtf, .sql, .txt, .xls, .zip

>**Background:**
>We’ve been using Directus v6 for a couple of years and I recently started a manual migration to v8.
>So I:
>* Manually recreated all of the Collections
>* Migrated the files by copying them to the correct location
>* Added the correct database entries into directus_files (including a private hash)
>* Corrected a few of the content types (to match what v8 saves them in the database as) 
Still no luck :confused:  Has anyone created a guide yet on how to correctly perform a “legacy upgrade”?

Unfortunately this is a pretty serious hold up as I'm in the midst of a Client deadline, any help you can provide is much appreciated!

Thanks for the help, I'm really excited to move forward with the new version 😎 !!!

**Comment by allanleonardjr** [comment]
@rijkvanzanten - Any quick suggestions you can offer on workarounds I might implement myself so I can continue moving forward with the project? (Note: I am a developer, mainly React.js throughout the day however, I'm more than happy to make some local code adjustments if it will allow me to overcome this hump.... just need a little direction).
Again, I'm so thankful for the work you all are doing!

**Comment by rijkvanzanten** [comment]
It looks to me like it's trying to read the thumbnails for an image that doesn't have thumbnails. I _think_ it might be the `.bmp` files that fail, as those have a mimetype that starts with `image`, but might not have generated thumbnails 🤔 

The other file types you shared don't have a mimetype starting with `image` and are returned as `null` in there. If possible, could you confirm by temporarily removing the `.bmp` images? If that's the case, the fix would be to do a check in a mime type allow-list of the types that the api actually generates thumbnails for instead of generally mimetypes that start with image

**Comment by allanleonardjr** [comment]
Thanks @rijkvanzanten sorry I forgot to update the list of filetypes that I'm using. About 3 weeks ago I went through one by one and tested each file type and I found, like you, that the .bmp files appeared to be causing an issue so I converted them to .jpg.... however once I imported all the files into the console again the issue was back, then I opened this issue.

What I might do is temporarily modify the code to return `null` even for the `image` mimetypes just to keep things moving until I can pinpoint the issue. Would that break anything _(other than not showing thumbnails in the console)_?

Curious, where can I find the "allow-list" of types the api can generate thumbnails for?

**Comment by rijkvanzanten** [comment]
I believe it's jpeg/png/webp, but i'm not 100% sure (I'm not too familiar in v8s api myself 🙂 )

**Comment by allanleonardjr** [comment]
Ok thanks, I'll check it out.

**Comment by allanleonardjr** [comment]
@rijkvanzanten after further troubleshooting I found the source of my problem and fixed it! 

Essentially it came down to the `type` field in the `directus_files` table for a few records. 
When I tested all of the different file types a few weeks ago, I found the .bmp files to be problematic so I converted them to .jpg. However, somehow, their `type` field in the database got set to **`images/jpeg` instead of `image/jpeg`**. I corrected those records and now I'm back up and running, thanks!!!!!!!

![image](https://user-images.githubusercontent.com/16966049/90848469-f5394180-e33a-11ea-8ffc-51d1b4496cdb.png)

**Comment by reitsma** [comment]
I encountered the same issue after upgrade from 7.0.* to 8.8.1. After reading the replies above, I removed one by one the images that were not real images like an pdf, and a jp2 (?) file. After that, it worked again. I guess the GUI should prevent loading images that are not images in the first place. Perhaps that is not the case, did not try yet with the new GUI. I now have image/jpeg, image/png and image/gif in the files table. That works ok.
